src/module.rs

```diff
@@ -19,35 +19,37 @@ use wamr_sys::{
 #[allow(dead_code)]
 #[derive(Debug)]
 pub struct Module {
+    name: String,
     module: wasm_module_t,
     // to keep the module content in memory
     content: Vec<u8>,
     wasi_ctx: WasiCtx,
 }
 
 impl Module {
-    /// compile a module with the given wasm file path
+    /// compile a module with the given wasm file path, use the file name as the module name
     ///
     /// # Error
     ///
     /// If the file does not exist or the file cannot be read, an `RuntimeError::WasmFileFSError` will be returned.
     /// If the wasm file is not a valid wasm file, an `RuntimeError::CompilationError` will be returned.
     pub fn from_file(runtime: &Runtime, wasm_file: &Path) -> Result<Self, RuntimeError> {
+        let name = wasm_file.file_name().unwrap().to_str().unwrap();
         let mut wasm_file = File::open(wasm_file)?;
 
         let mut binary: Vec<u8> = Vec::new();
         wasm_file.read_to_end(&mut binary)?;
 
-        Self::from_buf(runtime, &binary)
+        Self::from_buf(runtime, &binary, name)
     }
 
-    /// compile a module int the given buffer
+    /// compile a module int the given buffer,
     ///
     /// # Error
     ///
     /// If the file does not exist or the file cannot be read, an `RuntimeError::WasmFileFSError` will be returned.
     /// If the wasm file is not a valid wasm file, an `RuntimeError::CompilationError` will be returned.
-    pub fn from_buf(_runtime: &Runtime, buf: &[u8]) -> Result<Self, RuntimeError> {
+    pub fn from_buf(_runtime: &Runtime, buf: &[u8], name: &str) -> Result<Self, RuntimeError> {
```

Comment:
Should we check whether the name is registered and prevent the loading if it is registered?

And should we provide another API to find the module from the registered name?

Reply:
for now, it is a name stored only in rust layer 😄

Reply:
Yes, but I wonder whether it meets the requirement, e.g. different modules are bound to a same name, and when runwasi retrieves the module with a name, it may get different module?

Reply:
@Mossaka would u mind giving us details about the requirement of "module name"?

Reply:
depends on https://github.com/bytecodealliance/wasm-micro-runtime/pull/3254